File: 01_Language/01_Functions/python/checkdate.py
# coding: utf-8

import datetime


def checkdate(m, d, y):
    y, m, d = map(int, (y, m, d))
    try:
        datetime.date(y, m, d)
        return True
    except ValueError:
        return False


if __name__ == '__main__':
    print(checkdate(12, 31, 2000))
    print(checkdate(2, 29, 2001))

File: 01_Language/01_Functions/js/checkdate.js
// https://locutus.io/php/datetime/checkdate/index.html
function checkdate(m, d, y) {
    return m > 0 && m < 13 &&
        y > 0 && y < 32768 &&
        d > 0 && d <= (new Date(y, m, 0)).getDate()
}

console.log(checkdate(12, 31, 2000))
console.log(checkdate(2, 29, 2001))
